frontend/packages/metal3-plugin/src/components/host-detail.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+
+import { Row, Col } from 'patternfly-react';
+
+import { referenceForModel } from '@console/internal/module/k8s';
+import { DetailsPage } from '@console/internal/components/factory';
+import { navFactory } from '@console/internal/components/utils';
+import { units } from '@console/internal/components/utils/units';
+import { getName } from '@console/shared';
+
+import { BaremetalHostModel } from '../models';
+import MachineCell from './machine-cell';
+import {
+  getHostNICs,
+  isHostOnline,
+  getHostDescription,
+  getHostBMCAddress,
+  getHostCPUs,
+  getHostRAM,
+  getHostTotalStorageCapacity,
+} from '../selectors';
+
+type BaremetalHostDetailPageProps = {
+  namespace: string;
+  name: string;
+  match: any;
+};
+
+const BaremetalHostDetails = ({ obj }) => {
+  const nics = getHostNICs(obj);
+  const online = isHostOnline(obj);
+  const ips = nics.map((nic) => nic.ip).join(', ');
+
+  const statusClasses = classNames({
+    fa: true,
+    'co-icon-and-text__icon': online === true,
+    'fa-refresh': true,
+  });
+
+  const totalCapacity = units.humanize(
+    // The value from the selector is in GB
+    getHostTotalStorageCapacity(obj) * 1024 ** 3,
```

Comment:
Existing code, but it would be good if this helper specified the unit like `getHostTotalStorageCapacityGB`. If there's no unit, I would have assumed bytes.